static/js/stream_list.js

```diff
@@ -524,14 +524,20 @@ export function update_stream_sidebar_for_narrow(filter) {
         $stream_li.addClass("active-filter");
     }
 
+    // This makes sure we stay in zoomed in state when a stream updated
+    // unless the user switches to a different stream/narrow.
+    const was_zoomed_in = topic_zoom.is_zoomed_in() && !narrow_changed;
     topic_zoom.clear_topics();
     topic_list.rebuild($stream_li, stream_id);
+    if (was_zoomed_in) {
+        topic_zoom.zoom_in();
+    }
```

Comment:
Do you understand how this used to work? I feel like it would be cleaner to just not clear the zoomed state.

Reply:
Ok, I pushed a new implementation which avoids clearing of zoomed state altogether.